lib/services/local/presence.go

```diff
@@ -248,6 +248,13 @@ func (s *PresenceService) UpsertNode(ctx context.Context, server types.Server) (
 	if n := server.GetNamespace(); n != apidefaults.Namespace {
 		return nil, trace.BadParameter("cannot place node in namespace %q, custom namespaces are deprecated", n)
 	}
+
+	// Use labels from discovered resources instead of ones reported by discovered ec2 instances
+	server, err := s.filterEC2Labels(ctx, server)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
We shouldn't be checking this in the local backend directly. Instead, should make sure that DiscoveredServer resource is cached like other resources and have the auth server check the local cache for the node heartbeats.

cc @fspmarshall